src/main/java/net/geforcemods/securitycraft/screen/BlockPocketManagerScreen.java

```diff
@@ -53,6 +61,29 @@ protected void drawGuiContainerForegroundLayer(int mouseX, int mouseY)
 		String translation = ClientUtils.localize(SCContent.BLOCK_POCKET_MANAGER.get().getTranslationKey());
 
 		font.drawString(translation, xSize / 2 - font.getStringWidth(translation) / 2, 6, 4210752);
+
+		if (!te.enabled)
+		{
+			font.drawString(ClientUtils.localize("gui.securitycraft:blockPocketManager.youNeed"), 7, 83, 4210752);
+
+			font.drawString((size - 2) * (size - 2) * 6 + "", 42, 100, 4210752);
+			minecraft.getItemRenderer().renderItemAndEffectIntoGUI(BLOCK_POCKET_WALL, 25, 96);
+
+			font.drawString((size - 2) * 12 - 1 + "", 94, 100, 4210752);
+			minecraft.getItemRenderer().renderItemAndEffectIntoGUI(REINFORCED_CHISELED_CRYSTAL_QUARTZ, 77, 96);
+
+			font.drawString("8", 147, 100, 4210752);
+			minecraft.getItemRenderer().renderItemAndEffectIntoGUI(REINFORCED_CRYSTAL_QUARTZ_PILLAR, 130, 96);
+
+			if(mouseX >= guiLeft + 23 && mouseX < guiLeft + 48 && mouseY >= guiTop + 93 && mouseY < guiTop + 115)
+				renderTooltip(new ItemStack(SCContent.BLOCK_POCKET_WALL.get()), mouseX - guiLeft, mouseY - guiTop);
+
+			if(mouseX >= guiLeft + 75 && mouseX < guiLeft + 100 && mouseY >= guiTop + 93 && mouseY < guiTop + 115)
+				renderTooltip(new ItemStack(SCContent.REINFORCED_CRYSTAL_QUARTZ_PILLAR.get()), mouseX - guiLeft, mouseY - guiTop);
+
+			if(mouseX >= guiLeft + 128 && mouseX < guiLeft + 153 && mouseY >= guiTop + 93 && mouseY < guiTop + 115)
+				renderTooltip(new ItemStack(SCContent.REINFORCED_CHISELED_CRYSTAL_QUARTZ.get()), mouseX - guiLeft, mouseY - guiTop);

```

Comment:
There are _still_ 3 instances here where you create a new ItemStack each tick (on hover, but still).

Reply:
Sorry, I didn't see these instances and I also didn't look at the updated review above, so it took me some time.